lib/wibox/layout/stack.lua

```diff
@@ -0,0 +1,166 @@
+---------------------------------------------------------------------------
+-- @author Emmanuel Lepage Vallee
+-- @copyright 2016 Emmanuel Lepage Vallee
+-- @release @AWESOME_VERSION@
+-- @classmod wibox.layout.stack
+---------------------------------------------------------------------------
+
+local base  = require("wibox.widget.base" )
+local fixed = require("wibox.layout.fixed")
+local table = table
+local pairs = pairs
+local floor = math.floor
+local util  = require("awful.util")
+
+local stack = {mt={}}
+
+--- Layout a fixed layout. Each widget gets just the space it asks for.
+-- @param layout The layout you are modifying.
+-- @param context The context in which we are drawn.
+-- @tparam number width The available width.
+-- @tparam number height The available height.
+-- @name layout
+-- @class function
+
+--- Get all direct children widgets
+-- @param layout The layout you are modifying.
+-- @return a list of all widgets
+-- @name get_children
+-- @class function
+
+--- Add some widgets to the given fixed layout
+-- @param layout The layout you are modifying.
+-- @tparam widget ... Widgets that should be added (must at least be one)
+-- @name add
+-- @class function
+
+--- Set a widget at a specific index, replace the current one
+-- @tparam number index A widget or a widget index
+-- @param widget2 The widget to take the place of the first one
+-- @treturn boolean If the operation is successful
+-- @name set
+-- @class function
+
+--- Remove a widget from the layout
+-- @tparam index The widget index to remove
+-- @treturn boolean index If the operation is successful
+-- @name remove
+-- @class function
+
+--- Fit the fixed layout into the given space
+-- @param layout The layout you are modifying.
+-- @param context The context in which we are fit.
+-- @tparam number orig_width The available width.
+-- @tparam number orig_height The available height.
+-- @name fit
+-- @class function
```

Comment:
Wait, this inherits `:fit()` from fixed? That makes no sense at all, does it?
Could you stack e.g. an imagebox and a textbox ontop of each other, add some other widget to the left/right of this and put all of that in a fixed layout? The idea is to verify that each widget gets a sane amount of space assigned (due to the inherited `:fit()`, the stack should be too wide).